Fixed bug in position refinement parameter parsing

daurer opened the pull request:
I found that in the current version of the position correction engine, that defaults and user-provided parameters are not updated properly. Position correction only works if all posref parameters are provided in the configuration. 

I've made a quick fix for this. @pierrethibault is there are more elegant way of solving this?

I've also included new tests.

daurer committed:
Fixed typos

daurer committed:
Added tests for position correction

daurer committed:
Fixed bug in parsing posref parameters

daurer committed:
Removed unnecessary import

bjoernenders commented:
This is a general issue. It looks like all subclasses read the parameters again. Subclasses shouldn't need to read the parameter tree. Needs to be a separate issue. Thanks for finding this bug Benedikt!

daurer committed:
Move parameter update into init() and remove unnecessary update in DM

daurer committed:
Remove unnecessary param update in ML

daurer commented:
Removed the second reading of the parameter tree in DM and ML. But this does not fix the issues with the position refinement subtree, which needs to read sepearately. I have moved that part now into the init() of the PositionCorrection engine.

daurer committed:
Added reference to open issue #256
